bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/api/ShellyHttpApi.java

```diff
@@ -489,14 +495,17 @@ private String request(String uri) throws IOException {
                 timeoutsRecovered++;
                 logger.debug("Shelly API timeout recovered");
             } catch (IOException e) {
+                String message = e.getMessage();
```

Comment:
It appears this logic is used in multiple places. Please create a method for it.

Reply:
yep, tried to fixed them all (except some logger.xxx. which is able to handle null)